File: scrape.py
import urllib.request, re, json
req = urllib.request.Request('https://maplestory.nexon.com/News/CashShop', headers={'User-Agent': 'Mozilla/5.0'})
try:
    html = urllib.request.urlopen(req).read().decode('utf-8')
    images = re.findall(r'<img[^>]*src=[\'\"]([^\'\"]*?(?:png|jpg|jpeg|gif))[\'\"]', html)
    print(json.dumps(list(set(images)), indent=2))
except Exception as e:
    print("Error:", e)

File: scrape.js
const https = require('https');

https.get('https://maplestory.nexon.com/News/CashShop', (res) => {
  let data = '';
  res.on('data', (chunk) => {
    data += chunk;
  });
  res.on('end', () => {
    const regex = /<img[^>]+src=["']([^"']+\.(?:png|jpg|jpeg|gif))["']/g;
    let match;
    const urls = new Set();
    while ((match = regex.exec(data)) !== null) {
      urls.add(match[1]);
    }
    console.log(Array.from(urls).join('\n'));
  });
}).on('error', (err) => {
  console.log('Error: ' + err.message);
});
